src/palette/elements/Pill/Pill.tsx

```diff
@@ -1,38 +1,70 @@
-import React from "react"
+import OpaqueImageView from "lib/Components/OpaqueImageView/OpaqueImageView"
+import React, { ReactNode } from "react"
 import { GestureResponderEvent, TouchableOpacity } from "react-native"
+import styled from "styled-components/native"
 import { useTheme } from "../../Theme"
 import { Flex, FlexProps } from "../Flex"
 import { Text } from "../Text"
 
+const SIZES = {
+  xxs: {
+    height: 25,
+    typeSize: "12",
+  },
+  xs: {
+    height: 35,
+    typeSize: "13",
+  },
+  sm: {
+    height: 45,
+    typeSize: "16",
+  },
+}
+
 interface PillProps extends FlexProps {
+  selected?: boolean
+  rounded?: boolean
+  size?: "xxs" | "xs" | "sm"
+  imageUrl?: string
+  icon?: ReactNode
+  iconPosition?: "left" | "right"
   onPress?: (event: GestureResponderEvent) => void
-  // if no variant is passed defaulting to textSquare
-  variant?: "textRound"
-  active?: boolean
 }
 
-export const Pill: React.FC<PillProps> = ({ children, style, variant, active = true, onPress, ...other }) => {
-  const { colorV3, space } = useTheme()
-  const borderColor = active ? colorV3("black60") : colorV3("black15")
-  const borderRadius = variant === "textRound" ? 50 : 0
+export const Pill: React.FC<PillProps> = ({
+  children,
+  size = "xxs",
+  selected,
+  rounded,
+  icon,
+  iconPosition = "left",
+  imageUrl,
+  onPress,
+  ...other
+}) => {
+  const { color, space } = useTheme()
```

Comment:
this should be `colorV3` isn't it? 🤔
```suggestion
  const { colorV3, space } = useTheme()
```